tests/unit/lms/validation/_base_test.py

```diff
@@ -30,3 +33,48 @@ def test_it_fails_without_json_content_type_header(self, content_type):
 
         with pytest.raises(HTTPUnsupportedMediaType):
             self.ExampleSchema(request).parse()
+
+
+class TestRequestsResponseSchema:
+    class MySchema(RequestsResponseSchema):
+        test_field = fields.Str(required=True)
+
+    def test_it_returns_the_parsed_data_if_the_response_is_valid(self):
+        response = factories.requests.Response(json_data={"test_field": "test_value"})
+        schema = self.MySchema(response)
+
+        parsed_data = schema.parse()
+
+        assert parsed_data == {"test_field": "test_value"}
+
+    @pytest.mark.parametrize(
+        "response_body,validation_errors",
+        [
+            pytest.param(
+                "{}",
+                {"test_field": ["Missing data for required field."]},
+                id="Valid JSON but doesn't match the schema",
+            ),
+            pytest.param(
+                "foo",
+                {"_schema": ["response doesn't have a valid JSON body"]},
+                id="Not valid JSON at all",
+            ),
+        ],
+    )
+    def test_it_raises_ExternalRequestError_if_the_response_is_invalid(
+        self, response_body, validation_errors
+    ):
+        response = factories.requests.Response(
+            raw=response_body, request=sentinel.request
+        )
+        schema = self.MySchema(response)
+
+        with pytest.raises(ExternalRequestError) as exc_info:
+            schema.parse()
+
+        exc = exc_info.value
+        assert isinstance(exc, ExternalRequestError)
+        assert exc.request == sentinel.request
+        assert exc.response == response
+        assert exc.validation_errors == validation_errors
```

Comment:
`RequestsResponseSchema` had no unit tests :fearful: It actually does quite a few things since it's responsible for raising if the JSON-parsing fails as well as if the schema validation fails, and it's now responsible for setting `request`, `response` and `validation_errors`. So I added a couple of tests